loadgen/contract/agent-prepare-loadgen.js

```diff
@@ -406,42 +407,62 @@ export default async function startAgent({
     },
   );
 
-  return E.when(vaultManager, async (vaultManagerPresence) => {
-    const collateralTokenPetname =
-      fallbackCollateralToken || fallbackTradeToken;
-
-    if (vaultManagerPresence) {
-      return { vaultTokenKit: tokenKit, ammTokenKit: tokenKit };
-    } else if (collateralTokenPetname) {
-      // Make sure the finder knows about all purses by finding the
-      // LGT purse we created
-      await purseFinder.find({ brandPetname: tokenBrandPetname });
-
-      const { kit: vaultTokenKit } = E.get(
-        purseFinder.find({
-          brandPetname: collateralTokenPetname,
-          existingOnly: true,
-        }),
-      );
-
-      let ammTokenKit = vaultTokenKit.then((value) => value || tokenKit);
-      if (
-        (fallbackTradeToken && fallbackTradeToken !== collateralTokenPetname) ||
-        !(await fundingResult)
-      ) {
-        ({ kit: ammTokenKit } = E.get(
+  return E.when(
+    Promise.all([vaultManager, vaultFactoryPublicFacet]),
+    async ([vaultManagerPresence, vaultFactory]) => {
+      const collateralTokenPetname =
+        fallbackCollateralToken || fallbackTradeToken;
+
+      /** @type {ERef<import('../types.js').VaultCollateralManager | null>} */
```

Comment:
looks like it'll be null if `!vaultManagerPresence` and `ERef<null>` if `getCollateralManager` fails.
```suggestion
      /** @type {ERef<import('../types.js').VaultCollateralManager | null> | null} */
```

Reply:
`ERef<T>` is equivalent to `ERef<T> | T`

Reply:
oh right